### PR TITLE
Tweaks for block fill algorithm

### PR DESCRIPTION
**Pull request by emlowe**
Tweaks to the block fill algorithm

- sets a maximum number of spendbundles to consider to keep the time limited
- sets a threshold of 99.8% full
- Tries to fit as many items as possible in the block given within the above constraints

Not quite sure how to write a test - currently WIP draft

### CODE REVIEW

File: chia/full_node/mempool.py [hunk 1]
```diff
@@ -392,9 +392,12 @@ def create_bundle_from_mempool_items(
         eligible_coin_spends = EligibleCoinSpends()
         coin_spends: List[CoinSpend] = []
         sigs: List[G2Element] = []
+        limit = 750  # Maximum number of spendbundles to consider - should be based on the BLOCK_SIZE_LIMIT_FACTOR
```
Comment (arvidn) [inline chia/full_node/mempool.py:395]: with 100% fill rate, I expect we can fit closer to 1000 transactions in a block

File: chia/full_node/mempool.py [hunk 2]
```diff
@@ -412,13 +415,15 @@ def create_bundle_from_mempool_items(
...
                 ):
-                    break
+                    continue  # Keep looking for smaller items to fill the block
                 coin_spends.extend(unique_coin_spends)
                 additions.extend(unique_additions)
                 sigs.append(item.spend_bundle.aggregated_signature)
                 cost_sum += item_cost
                 fee_sum += fee
                 processed_spend_bundles += 1
+                if cost_sum / self.mempool_info.max_block_clvm_cost >= 0.998:  # 99.8% full seems a reasonable target
```
Comment (arvidn) [inline chia/full_node/mempool.py:425]: I think a better model is to stop when the cost left to fill in the block is smaller than some absolute threshold. That way it doesn't depend on the fill rate

**Review by arvidn — COMMENTED** [review]
Amine's patch is more efficient, I think. But it's also more complicated: https://github.com/Chia-Network/chia-blockchain/pull/17346

**Comment by emlowe** [comment]
Closing in favour of #17346

**Pull request closed without merging**